packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts

```diff
@@ -143,11 +166,26 @@ export class BaseTypesVisitor<TRawConfig extends RawTypesConfig = RawTypesConfig
     const originalNode = parent[key] as ObjectTypeDefinitionNode;
     const optionalTypename = this.config.nonOptionalTypename ? '__typename' : '__typename?';
     const allFields = [...(this.config.addTypename ? [indent(`${optionalTypename}: '${node.name}',`)] : []), ...node.fields];
-    const interfaces = originalNode.interfaces && node.interfaces.length > 0 ? originalNode.interfaces.map(i => this.convertName(i)).join(' & ') + (allFields.length ? ' & ' : '') : '';
+    const { type } = this._parsedConfig.declarationKind;
+
+    const buildInterfaces = () => {
+      if (!originalNode.interfaces || !node.interfaces.length) {
+        return '';
+      }
+
+      const interfaces = originalNode.interfaces.map(i => this.convertName(i));
+
+      if (type === 'interface') {
+        return ' extends ' + interfaces.join(', ') + (allFields.length ? ' ' : ' {}');
```

Comment:
Required a stand alone function to differentiate between interface extension and type unions.